### PR TITLE
Update neomerx/jsonapi to ^3.0.0

### LINKED ISSUE #57: Upgrade to neomerx/jsonapi ^2.0
We are currently using ^1.0 which is not the latest/current version.

### PR DESCRIPTION
**Pull request by dakota**
This closes #57 and upgrades crud-json-api to use the latest version of neomerx/jsonapi. This also changes the api to use (JSON:API v1.1)[https://jsonapi.org/format/1.1]

### CODE REVIEW

**Comment by bravo-kernel** [comment]
Wow, this is amazing, thank you so much. Do you need me to merge this once it is done or do you have merge permission yourself?

**Comment by dakota** [comment]
I have merge permission :)

I think it is done, all tests pass, and my own application seems to be working with it. I would appreciate a 2nd pair of eyes looking at the code before it gets merged if you've got the time.

**Comment by bravo-kernel** [comment]
Fair enough, I will try to look at it later today and report back here.

File: tests/TestCase/View/JsonApiViewTest.php [hunk 1]
```diff
@@ -281,18 +287,18 @@ public function testEncodeWithoutSchemas()
...
-            '_withJsonApiVersion' => true
-        ]);
-        $expectedVersionArray = [
-            'jsonapi' => [
-                'version' => '1.0'
-            ]
-        ];
-
-        $this->assertArraySubset($expectedVersionArray, json_decode($view->render(), true));
+//        $countries = TableRegistry::get('Countries')->find()->all();
```
Comment (bravo-kernel) [inline tests/TestCase/View/JsonApiViewTest.php]: Should this not be removed?

**Comment by bravo-kernel** [comment]
Besides the commented-out code it all looks good to me so feel free to merge and create a new release.

**Comment by bravo-kernel** [comment]
👏

**Comment by bravo-kernel** [comment]
Awesome job mr. @dakota 👏

**Pull request merged**